rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -2876,7 +2876,15 @@ int LuaUnsyncedCtrl::MarkerErasePosition(lua_State* L)
 	                 luaL_checkfloat(L, 2),
 	                 luaL_checkfloat(L, 3));
 
-	inMapDrawer->SendErase(pos);
+	/* Argument 4 is going to be radius
+	 * after some future refactoring. */
+
+	const bool onlyLocal = (lua_isnumber(L, 5)) ? bool(luaL_optnumber(L, 5, 0)) : luaL_optboolean(L, 5, false);
```

Comment:
Is there a reason this check is not just `luaL_optboolean(L, 5, false)` ?

Reply:
Yes, it's for consistency with how this arg works for the other marker functions (see e.g. line 2858). I agree it's lame though, perhaps another commit afterwards could change them all to be just optboolean.

Reply:
Yeah it should, the code looks confusing for no apparent reason here.